### PR TITLE
テストコード＆実装

### PR DESCRIPTION
**Pull request by sutefu23**
引数のチェックは値オブジェクトを使って実装しました。
仮実装までをmasterブランチでやって、テスト駆動でかいたものをhirakawa/24レポジトリで実装しました。

```npm install```
```npm run test```

## Summary by CodeRabbit

- **Test**
  - Adjusted the code coverage collection to include different files.
  - Improved various test cases for better accuracy and added a new test case for division by zero.
- **Refactor**
  - Updated the `NumberParams` class and arithmetic functions for better error handling and validation.
    - The arithmetic functions now handle specific edge cases and return appropriate messages.
    - The `NumberParams` class now validates input numbers more effectively.
- **New Feature**
  - Added a `values` property to the `NumberParams` class to store an array of numbers.

### CODE REVIEW

File: src/script.test.ts [hunk 1]
```diff
@@ -18,115 +16,151 @@ describe("パラメーターのテスト", () => {
...
-      5, 5, 5, 5, 5, 5,
+      5, 5, 5, 5, 5,
     ];
     expect(param.length).toBeLessThanOrEqual(30); //30以下
 
-    const actual = new NumberParams(param);
+    const actual = () => new NumberParams(param);
     expect(actual).not.toThrowError();
   });
 });
```
Comment (YamazakiYusuke) [inline src/script.test.ts:32]: パラーメーターの要件が共通なので値オブジェクトにされたのですね。流石です！
かなり、重複が減りますね、

File: src/script.ts [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
+
+export const multiply = (params: NumberParams) => {
+  const result = params.values.reduce((acc, cur) => acc * cur, 1);
+  return result > 1000 ? "big big number" : result;
 };
 
 export const divide = (params: NumberParams) => {
-  return 0;
+  if (params.values.some((param) => param === 0))
+    throw new Error("0で割り算をすることは出来ません。");
```
Comment (YamazakiYusuke) [inline src/script.ts:32]: 👍

File: src/script.ts [hunk 3]
```diff
@@ -1,18 +1,42 @@
 export class NumberParams {
-  constructor(numbers: number[]) {}
-}
-export const multiply = (params: NumberParams) => {
-  return 0;
-};
+  public values: number[];
```
Comment (YamazakiYusuke) [inline src/script.ts]: 細かい指摘で恐縮ですが、NumberParamsのインスタンス変数valuesはpublicよりreadonlyにするのが良いかと思います。

Comment (sutefu23) [reply src/script.ts]: いえいえ。うーん。確かにValueObjectだとそうですね。。変えます！

Comment (sutefu23) [reply src/script.ts]: @YamazakiYusuke 
修正しました！

File: src/script.ts [hunk 4]
```diff
@@ -1,18 +1,42 @@
...
-  constructor(numbers: number[]) {}
-}
-export const multiply = (params: NumberParams) => {
-  return 0;
-};
+  public readonly values: number[];
+  constructor(numbers: number[]) {
+    numbers.forEach((num, index) => {
+      if (typeof num !== "number")
+        throw new Error(`${index + 1}番目の引数が数値ではありません`);
```
Comment (hiraaaken) [inline src/script.ts:6]: ここのエラーメッセージユーザーに親切でいいですね👌

**Review by hiraaaken — APPROVED** [review]
こちらも確認しました！LGTMです！
仮実装から本実装されたとのことで、TDDによってどう変わったがわかりやすくて勉強になりました🙆‍♂️
自分も引数をオブジェクトにする方法を取りましたが、TDDだといつもなら実装をやめてしまう段階よりもさらに深くコードについて考えられるような気がしました！（最大限重複を減らすにはどうしたらいいか…とか）
